tests/PHPCR/Tests/Util/QOM/Sql2ScannerTest.php

```diff
@@ -124,6 +124,60 @@ public function testSQLEscapedStrings2()
         $this->expectTokensFromScanner($scanner, $expected);
     }
 
+    public function testSquareBrackets()
+    {
+        $sql = 'WHERE ISSAMENODE(file, ["/home node"])';
+
+        $scanner = new Sql2Scanner($sql);
+        $expected = [
+            'WHERE',
+            'ISSAMENODE',
+            '(',
+            'file',
+            ',',
+            '[',
+            '"/home node"',
+            ']',
```

Comment:
Yeah now the bracketed expression is no longer one but 3 parts. So I would propose everything that is in brackets to be parsed as one. Token, if that is the desired behavior. The question then becomes what if the query string inside is invalid. Something like

```sql
WHERE ISSAMENODE(file, ["/home node])
```
with an unclosed quote.

Reply:
I have updated the merge request with a bracket parser. And the invalid nodes are allowed but will be parsed as one token.